Final removal of legacy input parameters

loganharbour opened the pull request:
Closes #19439

loganharbour committed:
Final removal of legacy input parameters

refs #19439

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1012584/) on 50985e7 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/20779/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1012609/) on 50985e7 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>e9e9d4</th>
      <th colspan=4>#20779 50985e</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>82.56%</td>
      <td>82.56%</td>
      <td>-0.01%</td>
      <td>85.71%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>74150</td>
      <td>74102</td>
      <td>-48</td>
      <td>6</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15660</td>
      <td>15658</td>
      <td>-2</td>
      <td>1</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20779/coverage_diff/framework.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20779/coverage/framework)


## Modules coverage

### Fluid properties

<table>
  <thead>
    <tr>
      <th></th>
      <th>e9e9d4</th>
      <th colspan=4>#20779 50985e</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.56%</td>
      <td>85.35%</td>
      <td>-0.22%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>5542</td>
      <td>5528</td>
      <td>-14</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>935</td>
      <td>949</td>
      <td>+14</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20779/coverage_diff/fluid_properties.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20779/coverage/fluid_properties)

### Heat conduction

<table>
  <thead>
    <tr>
      <th></th>
      <th>e9e9d4</th>
      <th colspan=4>#20779 50985e</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>89.58%</td>
      <td>88.80%</td>
      <td>-0.78%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>2992</td>
      <td>2966</td>
      <td>-26</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>348</td>
      <td>374</td>
      <td>+26</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20779/coverage_diff/heat_conduction.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20779/coverage/heat_conduction)

### Misc

<table>
  <thead>
    <tr>
      <th></th>
      <th>e9e9d4</th>
      <th colspan=4>#20779 50985e</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>60.11%</td>
      <td>57.10%</td>
      <td>-3.01%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>220</td>
      <td>209</td>
      <td>-11</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>146</td>
      <td>157</td>
      <td>+11</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20779/coverage_diff/misc.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20779/coverage/misc)

### Navier stokes

<table>
  <thead>
    <tr>
      <th></th>
      <th>e9e9d4</th>
      <th colspan=4>#20779 50985e</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>83.68%</td>
      <td>80.04%</td>
      <td>-3.64%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>8419</td>
      <td>8053</td>
      <td>-366</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>1642</td>
      <td>2008</td>
      <td>+366</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20779/coverage_diff/navier_stokes.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20779/coverage/navier_stokes)

### Peridynamics

<table>
  <thead>
    <tr>
      <th></th>
      <th>e9e9d4</th>
      <th colspan=4>#20779 50985e</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>79.11%</td>
      <td>78.28%</td>
      <td>-0.83%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>3037</td>
      <td>3005</td>
      <td>-32</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>802</td>
      <td>834</td>
      <td>+32</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20779/coverage_diff/peridynamics.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20779/coverage/peridynamics)

### Phase field

<table>
  <thead>
    <tr>
      <th></th>
      <th>e9e9d4</th>
      <th colspan=4>#20779 50985e</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>90.21%</td>
      <td>89.81%</td>
      <td>-0.40%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>12039</td>
      <td>11986</td>
      <td>-53</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>1307</td>
      <td>1360</td>
      <td>+53</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20779/coverage_diff/phase_field.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20779/coverage/phase_field)

### Porous flow

<table>
  <thead>
    <tr>
      <th></th>
      <th>e9e9d4</th>
      <th colspan=4>#20779 50985e</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>96.28%</td>
      <td>96.12%</td>
      <td>-0.16%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>10533</td>
      <td>10516</td>
      <td>-17</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>407</td>
      <td>424</td>
      <td>+17</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20779/coverage_diff/porous_flow.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20779/coverage/porous_flow)

### Stochastic tools

<table>
  <thead>
    <tr>
      <th></th>
      <th>e9e9d4</th>
      <th colspan=4>#20779 50985e</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>89.84%</td>
      <td>89.60%</td>
      <td>-0.24%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>4432</td>
      <td>4420</td>
      <td>-12</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>501</td>
      <td>513</td>
      <td>+12</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20779/coverage_diff/stochastic_tools.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20779/coverage/stochastic_tools)

### Tensor mechanics

<table>
  <thead>
    <tr>
      <th></th>
      <th>e9e9d4</th>
      <th colspan=4>#20779 50985e</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>88.90%</td>
      <td>88.69%</td>
      <td>-0.21%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>26171</td>
      <td>26108</td>
      <td>-63</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>3268</td>
      <td>3331</td>
      <td>+63</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20779/coverage_diff/tensor_mechanics.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20779/coverage/tensor_mechanics)

### Thermal hydraulics

<table>
  <thead>
    <tr>
      <th></th>
      <th>e9e9d4</th>
      <th colspan=4>#20779 50985e</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>90.84%</td>
      <td>89.80%</td>
      <td>-1.04%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>11593</td>
      <td>11460</td>
      <td>-133</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>1169</td>
      <td>1302</td>
      <td>+133</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20779/coverage_diff/thermal_hydraulics.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20779/coverage/thermal_hydraulics)

### Xfem

<table>
  <thead>
    <tr>
      <th></th>
      <th>e9e9d4</th>
      <th colspan=4>#20779 50985e</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>81.87%</td>
      <td>81.82%</td>
      <td>-0.05%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>6344</td>
      <td>6340</td>
      <td>-4</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>1405</td>
      <td>1409</td>
      <td>+4</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20779/coverage_diff/xfem.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20779/coverage/xfem)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/20779/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/20779/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/20779/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/20779/coverage/contact)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/20779/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/20779/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/20779/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/20779/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/20779/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/20779/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/20779/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/20779/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/20779/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/20779/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/20779/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/20779/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/20779/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/20779/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/20779/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/20779/coverage/richards)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/20779/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/20779/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/20779/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/20779/coverage/xfem)

</details>

## Warnings

- `framework` new line coverage rate 85.71% is less than the suggested 90.0%



This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1012609/) on 50985e7 wanted to post the following:


The following coverage requirement(s) failed:
- `thermal_hydraulics` coverage rate 89.80% is less than the required 90.0%

loganharbour committed:
Decrease thermal_hydraulics required coverage rate to 89%

This is needed because the removal of CheckLegacyParamsActions
removed coverage for objects parameter construction

refs #19439
